[ide-eclipse] updated guide

mvalla opened the pull request:
removed infinite loop warning, added hint in case bundles are not listed.
After @kaikreuzer suggestion removed git checkout of openhab2-addons and used Oomph installer option to get openhab2-addons

kaikreuzer commented:
Thanks for updating the IDE description.
Would it make sense to also directly remove the "checkout the `openhab2-addons` repository" step and rather use the IDE installer feature for this (as we discussed in the forum)?

mvalla commented:
I will do a test. However this means all openhab2-addons would be compiled+tested, making the first run even longer. 
It’s already very very long right now, much more than before bndtools migration.

Instead by checking out openhab2-addons  from command line we avoid this to be done by Eclipse except for the single bindings you are interred in.
It also avoids having problems in case any of the committed binding does not compile at a certaint time , and avoids the need to close manually all bundles projects in Eclipse.

kaikreuzer commented:
@mvalla As mentioned before, if the projects are closed, it all works without an issue and is fast. The benefit is that the setup does the git checkout and the project import, so it is much easier for the user as the documentation does not become fragmented ("do this in the IDE and that on the command line...").
Ideally, the IDE setup should import the projects and close them automatically. I tried to check for options in the Oomph installer, but unfortunately couldn't easily find a way to do so (if anyone has an idea, please say so!). As a workaround, I would suggest to add a line to the docs that the user closes all projects by right-clicking the "OH2 Add-ons" working set and selecting "Close Projects", which closes all of them at the same time. Afterwards, the user can open the ones he wants to work with and they immediately compile without errors.
Having then mentioned the "openHAB Add-ons" installation option, we should likewise mention Z-Wave and Zigbee options, because they work flawlessly as well.

Gozilla01 commented:
@kaikreuzer ,
suggestion.
- You can't delete all the old IDE versions (03-2019, 06-2019, etc.), create a new 07-2019 (same identical copy of the last 03-2019) so it's more intuitive.
- In my opinion these two pages are a bit mashed and partly obsolete.
https://deploy-preview-1006--openhab-docs-preview.netlify.com/docs/developer/ide/eclipse.html
https://deploy-preview-1006--openhab-docs-preview.netlify.com/docs/developer/legacy/ide.html#installing-the-eclipse-ide
Maybe @davidgraeff  can give a hand to @mvalla  to unify all in one page..
What do you think about it?

davidgraeff commented:
The second linked page ("legacy/ide.html") is not rendered on the doc homepage and is only there for reference.

The first linked page is what you currently need to do to setup an IDE. There's nothing obsolete, except the infinite loop warning. I do not see the Oomph setup as a viable alternative as long as all projects are automatically opened and therefore I will not recommend it or help to promote it ;)

Apart from that I think it is very important that we have a guide for Eclipse that explains all the single steps to perform. Ooomph is hiding too much behind the scenes and there is no documentation anywhere on how to maintain the existing oomph script. That was also discussed in the community.

kaikreuzer commented:
The Oomph setup is the official way to get an IDE set up with everything that is needed (especially code formatters, etc.). It is imho no option to promote any other way right now and all of those should only be mentioned for people that know what they are doing and what they will be lacking in that case.

openhab-5iver commented:
> As a workaround, I would suggest to add a line to the docs that the user closes all projects by right-clicking the "OH2 Add-ons" working set and selecting "Close Projects", which closes all of them at the same time. 

Could we just remove the ProjectsImportTask setup task for add-ons and have users manually import the projects that they are interested in?

kaikreuzer commented:
Would again be more difficult to explain where to find them and what to do. I hope to find a way to automatically close them, this would be ideal.

mvalla commented:
I updated the guide. Now openhab2-addons is imported via Oomph setup. I tested it and it took 45min on a i7 16GB RAM fiber connection, but it's not so much more than doing git checkout manually and then compile openhab2-addons. Added "close projects" hint. 
For me the guide now is much easier for new users and provides a stable way to setup the IDE.

mvalla commented:
It has been 10 days: what does it stop for this to be merged?

davidgraeff commented:
Northern hemisphere summer break I guess :D

Confectrician commented:
>Northern hemisphere summer break I guess :D

Confectrician-has-a-huge-workload-those-days-break, i guess. :D
I have postponed this one for me, because ther was some discussion ongoing.

From what i read, discussion is finished now?

Confectrician commented on developers/ide/eclipse.md:
```diff
@@ -14,40 +14,61 @@ For now follow these steps:
-1. Start openHAB from the IDE clicking "Run OSGi" or "Debug OSGi" (upper right of the `app.bndrun` dialog).
-   You can check openHAB is running going with your browser to: http://localhost:8080/paperui/
-1. Check the chosen binding is active in PaperUI -> Configuration -> Bindings
+    These tasks will personalize the IDE with openHAB code formatting tools, configurations and a demo app. Initial tasks will also download openHAB lastest SNAPSHOT, the  `openhab2-addons` (bundles) code and related maven dependencies.
```
One line per sentence please.

mvalla replied:
done

Confectrician commented on developers/ide/eclipse.md:
```diff
@@ -14,40 +14,61 @@ For now follow these steps:
-   You can check openHAB is running going with your browser to: http://localhost:8080/paperui/
-1. Check the chosen binding is active in PaperUI -> Configuration -> Bindings
+    These tasks will personalize the IDE with openHAB code formatting tools, configurations and a demo app. Initial tasks will also download openHAB lastest SNAPSHOT, the  `openhab2-addons` (bundles) code and related maven dependencies.
+(click bottom right button in the IDE for Progress).
```
This looks weird with the brackets and the fullstop behind it.
I am no natove speaker, so does anyone now if this is correct grammatically?

openhab-5iver replied:
TMK, this is not grammatically correct. Even if it was, it just looks ugly :smile:. The 'initial tasks' referred to are 'setup tasks' in terms of the Eclipse Oomph setup, which perform a git clone of the repositories. There is no 'download of the latest snapshot', which is a packaged distribution and has nothing to do with the IDE.

mvalla replied:
done

kaikreuzer commented on developers/ide/eclipse.md:
```diff
@@ -14,40 +14,61 @@ For now follow these steps:
+    Eclipse version to use is "Latest Release (2019-03)" (currently version `2019-06` does not support bndtools)
 
-   ![demo import](images/ide_setup_eclipse_import_demo.png)
+1. Under `Github Projects > openHAB` select `openHAB Development` and `openHAB Add-ons`
```
```suggestion
1. Under `Github Projects > openHAB` select `openHAB Development` and any desired option from `openHAB Add-ons` (includes all add-ons from openhab2-addons repo), `openHAB ZigBee Binding` or `openHAB Z-Wave Binding`.
```

kaikreuzer commented on developers/ide/eclipse.md:
```diff
@@ -14,40 +14,61 @@ For now follow these steps:
-    ![Bndtools](images/ide_setup_eclipse_bndtools.png)
-1. Click "Resolve": a window with the list of resolved bundles will be shown.
-    Click Finish and save the file.
+1. **It's important**, during the first Eclipse IDE launch, to leave Eclipse open until all openaHAB related initial Startup tasks / Download sources / Builds are completed. 
```
```suggestion
1. **It's important**, during the first Eclipse IDE launch, to leave Eclipse open until all openHAB related initial Startup tasks / Download sources / Builds are completed. 
```

kaikreuzer commented on developers/ide/eclipse.md:
```diff
@@ -14,40 +14,61 @@ For now follow these steps:
-## Develop a NEW binding with the IDE
+    ![startup tasks progress](./images/ide_setup_eclipse_startup_tasks.png)
+
+    When visible in the Package Explorer, select `OH2 Add-ons` and from right-click menu select "Close Projects": this will speed up a bit the first setup.
```
```suggestion
    When visible in the Package Explorer, select `OH2 Add-ons` and from right-click menu select "Close Projects": this significantly speeds up the setup.
```

kaikreuzer commented on developers/ide/eclipse.md:
```diff
@@ -14,40 +14,61 @@ For now follow these steps:
+    ![startup tasks progress](./images/ide_setup_eclipse_startup_tasks.png)
+
+    When visible in the Package Explorer, select `OH2 Add-ons` and from right-click menu select "Close Projects": this will speed up a bit the first setup.
+    Leave open only the binding project(s) you would like to work on.
```
```suggestion
    Re-open only the binding project(s) you would like to work on.
```

kaikreuzer commented on developers/ide/eclipse.md:
```diff
@@ -14,40 +14,61 @@ For now follow these steps:
+    When visible in the Package Explorer, select `OH2 Add-ons` and from right-click menu select "Close Projects": this will speed up a bit the first setup.
+    Leave open only the binding project(s) you would like to work on.
+
+    **It may take 30-45mins for all initial tasks to finsih, depending on internet connection and your computer speed.
```
```suggestion
    **It may take 30-45mins for all initial tasks to finish, depending on internet connection and your computer speed.
```

kaikreuzer commented on developers/ide/eclipse.md:
```diff
@@ -14,40 +14,61 @@ For now follow these steps:
+
+   ![startup tasks finished](./images/ide_setup_eclipse_tasks_finished.png)
+
+1. After all stasks are finished, in the Eclipse Package Explorer in `Other Projects > launch > app` project folder look for `app.bndrun` file.
```
```suggestion
1. After all tasks are finished, look for `app.bndrun` file in the Eclipse Package Explorer in `Other Projects > launch > app` project folder.
```

kaikreuzer commented on developers/ide/eclipse.md:
```diff
@@ -14,40 +14,61 @@ For now follow these steps:
+1. Save and click "Resolve": a window with the list of resolved bundles will be shown.
+    Click `Finish` and save the file.
+
+Now the IDE is ready to start openHAB with a minimum set of the openHAB core bindings, PaperUI and the selected binding you want to run/debug.
```
```suggestion
Now the IDE is ready to start openHAB with a minimum set of the openHAB core bindings, UIs and the selected binding you want to run/debug.
```

kaikreuzer commented on developers/ide/eclipse.md:
```diff
@@ -14,40 +14,61 @@ For now follow these steps:
+
+1. Start openHAB from the IDE clicking "Run OSGi" or "Debug OSGi" (upper right of the `app.bndrun` window).
+   You can check openHAB is running going with your browser to: http://localhost:8080/paperui/ (the last `/` is important!)
+1. Check the chosen binding is active in `PaperUI > Configuration > Bindings`
```
```suggestion
1. Check the chosen binding is active in `Paper UI > Configuration > Bindings`
```

mvalla commented:
@Confectrician , @kaikreuzer I made changes to solve all your suggestions and comments.

kaikreuzer commented:
@mvalla Looks as if you need to rebase, though - there are two files in conflict.

mvalla commented:
> @mvalla Looks as if you need to rebase, though - there are two files in conflict.

@kaikreuzer done rebase (at least it seems so). No conflicts now.

kaikreuzer commented:
Thanks @mvalla!
